includes/src/Plugin.php

```diff
@@ -0,0 +1,334 @@
+        $this->cf_api_url = 'https://api.cloudflare.com/client/v4/';
+
+        $this->hook = plugin_basename(EXNANO_MUCFDNSM_FILE);
+        $this->path = realpath(plugin_dir_path(EXNANO_MUCFDNSM_FILE));
+        $this->plugin_url = plugin_dir_url(EXNANO_MUCFDNSM_FILE);
+        $this->screen = 'settings_page_exn-wpmu-cf-dns-manager';
+    }
+
+    private function init_actions()
+    {
+        // unofficial constant: possible to disable nag notices
+        !\defined('DISABLE_NAG_NOTICES') && \define('DISABLE_NAG_NOTICES', true);
+
+        add_action('network_admin_edit_'.$this->plugin_page, [$this, 'settings_save']);
+        add_action('network_admin_notices', [$this, 'custom_notices']);
+
+        add_action('wp_insert_site', [$this, 'create_dns_record']);
+        // add_action( 'wp_delete_site', [$this, 'delete_dns_record' ] );
+        // do_action( 'wp_insert_site', WP_Site $new_site );
+
+        add_action(
+            'admin_enqueue_scripts',
+            function ($hook) {
+                $plugin_url = $this->plugin_url;
+                $version = \defined('EXNANO_MUCFDNSM_VERSION') ? EXNANO_MUCFDNSM_VERSION : date('ymdh');
+
+                if ($hook === $this->screen) {
+                    wp_enqueue_style($this->plugin_page.'-core', $plugin_url.'includes/admin/exnano.css', null, $version);
+                }
+            }
+        );
+
+        add_action('network_admin_menu', function () {
+            add_submenu_page(
+                $this->menu_parent, // Parent element
+                'Cloudflare DNS Manager', // Text in browser title bar
+                'CF DNS Manager', // Text to be displayed in the menu.
+                'manage_network_options', // Capability
+                $this->plugin_page, // Page slug, will be displayed in URL
+                [$this, 'settings_page'], // Callback function which displays the page
+            );
+        });
+
+        add_filter(
+            'network_admin_plugin_action_links_'.$this->hook,
+            function ($links) {
+                $new = [
+                    'exnwpmucfdnsmanager-settings' => sprintf('<a href="%s">%s</a>', network_admin_url($this->menu_parent.'/?page='.$this->plugin_page), __('Settings', 'exn-wpmu-cf-dns-manager')),
+                ];
+
+                return array_merge($new, $links);
+            }
+        );
+    }
+
+    public function settings_page()
+    {
+        include_once $this->path.'/includes/admin/settings.php';
+    }
+
+    public function settings_save()
+    {
+        check_admin_referer($this->nonce_key);
+
+        update_site_option('exn_cf_api_token', sanitize_text_field($_POST['exn_cf_api_token']));
+
+        $redirect = add_query_arg(
+            [
+            'page' => $this->plugin_page,
+            'updated' => true,
+            ],
+            network_admin_url($this->menu_parent)
+        );
+
+        wp_redirect($redirect);
+
+        exit;
+    }
+
+    /**
+     * gcustom_notices.
```

Comment:
Is the `gcustom_notices` intented to spell like this?

Reply:
typo